osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -211,6 +220,30 @@ private void scoreChanged(ValueChangedEvent<ScoreInfo?> score)
                 return;
             }
 
+            unstableRate = hitEvents.CalculateUnstableRate();
+            const double unstablerate_threshold = 90;
+            const double exponential_factor = -0.0116;
+            // Parameters for the calculation that scales the offset with respect to UR.
+
+            if (autoAdjustBeatmapOffset.Value) //Enters this case if global setting is checked in Menu.
+            {
+                if (hitEvents.Count >= 30) //Make sure that there is a minimum baseline of hits to make evaluation.
+                {
+                    if (unstableRate >= unstablerate_threshold)
+                    {
+                        Current.Value = lastPlayBeatmapOffset - (average * Math.Exp((double)(exponential_factor * (unstableRate - unstablerate_threshold))));
```

Comment:
I'm sure this makes good sense, but without attempting to understand the consequences, I'm curious if there's a reason not to always use this algorithm even when the user presses the adjust button.

Reply:
Fair. Might be good to include this kind of explanation inline to give more context.

Reply:
I think I made a comment that leads any reader to the discussion where the calculation is explained in more detail. I didn't want to write such long comments since they say less than the image.
If you mean to give context to the user, the game setting name is already quite long, I couldn't think of a better way to name the setting for now.